src/remote-generator.ts

```diff
@@ -0,0 +1,158 @@
+    const data = await res.json()
+    return data
+}
+
+async function compareToLocal() {
+    const localData = await JSON.parse(fs.readFileSync('./generated/hyperlock.tokenlist.json', 'utf8'))
+    const remoteData: ThrusterTokenInfo[] = await fetchRemoteData()
+    const localAddresses: string[] = localData.tokens.map((token: Partial<TokenInfo>) => token.address?.toLowerCase())
+
+    const missingAddresses = remoteData.filter(({tokenAddress}) => !localAddresses.includes(tokenAddress.toLowerCase()))
+    return missingAddresses
+}
+
+async function generateTokens(
+    network: Network,
+    remoteTokens: ThrusterTokenInfo[],
+) {
+    const tokens: TokenInfo[] = []
+
+    for (const token of remoteTokens) {
+        const newToken={
+            chainId: network,
+            address: token.tokenAddress,
+            name: token.tokenName,
+            symbol: token.tokenSymbol,
+            logoURI: `https://raw.githubusercontent.com/hyperlockfi/tokenlists/main/src/assets/images/tokens/${token.tokenAddress.toLowerCase()}.png`,
+            decimals: token.tokenDecimals
+        } as unknown as TokenInfo
+        if(!IGNORE_REMOTE[token.tokenAddress]) tokens.push(newToken)
+    }
+
+    return tokens
+}
+
+async function getIcons(newTokens: TokenInfo[]) {
+    const imageUrls = newTokens.map(({address}) => ({address: address, url: `https://raw.githubusercontent.com/ThrusterX/thruster-token-list/main/chains/81457/assets/${address}/token-logo.svg`}))
+    
+    const downloadPromises = imageUrls.map(({ address, url }) => {
+        return new Promise((resolve, reject) => {
+            const filePath = `./src/assets/images/tokens/${address}.svg`;
+            const file = fs.createWriteStream(filePath);
+
+            https.get(url, response => {
+                response.pipe(file);
+
+                file.on('finish', () => {
+                    file.close(() => resolve(`Downloaded and saved: ${address}`));
+                });
+
+                file.on('error', (err) => {
+                    fs.unlink(filePath, () => {}); // Attempt to delete file on error
+                    reject(`Error saving file ${address}: ${err}`);
+                });
+            }).on('error', (err) => {
+                fs.unlink(filePath, () => {}); // Attempt to delete file on error
+                reject(`Error downloading ${address}: ${err}`);
+            });
+        });
+    });
+
+    try {
+        const results = await Promise.all(downloadPromises);
+        console.log(results);
+    } catch (e) {
+        console.error('Error fetching images:', e);
+    }
+}
+
+const convertToPng = (newTokens: boolean) => {
+
+    if (!newTokens) return
+    // Directory containing the SVG files
+    const directory = 'src/assets/images/tokens'
+    
+    // Command to find SVG files, convert them, rename, and delete the original
+    const command = `
+    cd ${directory} &&
+    for file in *.svg; do
+      rsvg-convert -h 128 "$file" -o "\${file%.svg}.png"
```

Comment:
Could use a node library like sharp here (https://github.com/lovell/sharp) to avoid additional deps outside of npm install. But maybe rsvg-convert is used in other scripts

Reply:
yeah, rsvg-convert is recommended by one of the readme's